internal/signer/darwin/keychain/keychain.go

```diff
@@ -482,34 +523,24 @@ func (k *Key) getRSADecryptAlgor() (C.SecKeyAlgorithm, error) {
 			algorithms = rsaPKCS1v15Algorithms
 		}
 	default:
-		return 0, fmt.Errorf("unsupported algorithm %T", pub)
+		return UNKNOWN_SECKEY_ALGORITHM, fmt.Errorf("unsupported algorithm %T", pub)
 	}
-	return algorithms[k.hashFunc], nil
+	return algorithms[k.hash], nil
 }
 
 func (k *Key) getDecryptAlgorithm() (C.SecKeyAlgorithm, error) {
-	return k.getRSADecryptAlgor()
-}
-
-func (k *Key) checkDataSize(plaintext []byte) error {
-	pub := k.publicKeyRef
-	// Plaintext data must be smaller than the key's block size.
-	maxSize := uint64(C.SecKeyGetBlockSize(pub))
-	if uint64(len(plaintext)) > maxSize {
-		return fmt.Errorf("plaintext is too long")
-	}
-	return nil
+	return k.getRSADecryptAlgorithm()
 }
 
 func (k *Key) Encrypt(plaintext []byte) ([]byte, error) {
 	pub := k.publicKeyRef
+	algorithm, err := k.getEncryptAlgorithm()
+	if err != nil {
+		fmt.Printf("algorithm is unsupported. only RSA algorithms are supported. %v", err)
```

Comment:
Maybe wrap this error with `fmt.Errorf` and return here

Reply:
Note: I deleted the fmt and instead just returned nil and err and err will print out the detailed error message